File: HW1/F19_10601_HW1/reverse/reverse.java
import java.nio.file.Files;
import java.nio.file.Paths;
import java.io.IOException;
import java.io.File;
import java.util.List;
import java.util.Collections;

public class reverse{
	public static void main(String[] args){
		try{
			System.out.println(args.length);
			if(args.length < 2){
				System.out.println("Please give both input and output files to the program");
			}
			String input_filename = args[0];
			String output_filename = args[1];
			System.out.println(input_filename);
			System.out.println(output_filename);
			List<String> lines = Files.readAllLines(Paths.get(input_filename));
			Collections.reverse(lines);
            		Files.write(Paths.get(output_filename), lines);
        	}
		catch(IOException e){
            		e.printStackTrace();
        	} 
	}
}

File: HW1/F19_10601_HW1/reverse/reverse.py
import sys

def main():
	num_args = len(sys.argv)
	print("Number of commandline args passed: %s" % num_args)
	if num_args < 3:
		print("Please give all the commandline args in the order: filename, input_filename, output_filename respectively. Current args given: %s" % sys.argv)
	with open(sys.argv[1]) as f:
		data = f.read()
	
	print("d: %s" % d)
	d.reverse()
	if d:
		d = d[1:]
		p = '\n'.join(d)
	with open(sys.argv[2], 'w') as f:
                f.write(d)
	print("Task done")


if __name__ == "__main__":
	main()
